meshers/blocky/voxel_mesher_blocky.cpp

```diff
@@ -47,6 +47,285 @@ inline bool contributes_to_ao(const VoxelLibrary &lib, int voxel_id) {
+							const int append_index = arrays.colors.size();
+							arrays.colors.resize(arrays.colors.size() + vertex_count);
+							Color *w = arrays.colors.data() + append_index;
+							const Color modulate_color = voxel.get_color();
+
+							if (bake_occlusion) {
+
+								for (unsigned int i = 0; i < vertex_count; ++i) {
+									Vector3 v = side_positions[i];
+
+									// General purpose occlusion colouring.
+									// TODO Optimize for cubes
+									// TODO Fix occlusion inconsistency caused by triangles orientation? Not sure if worth it
+									float shade = 0;
+									for (unsigned int j = 0; j < 4; ++j) {
+										unsigned int corner = Cube::g_side_corners[side][j];
+										if (shaded_corner[corner]) {
+											float s = baked_occlusion_darkness * static_cast<float>(shaded_corner[corner]);
+											//float k = 1.f - Cube::g_corner_position[corner].distance_to(v);
+											float k = 1.f - Cube::g_corner_position[corner].distance_squared_to(v);
+											if (k < 0.0) {
+												k = 0.0;
+											}
+											s *= k;
+											if (s > shade) {
+												shade = s;
+											}
+										}
+									}
+									const float gs = 1.0 - shade;
+									w[i] = Color(gs, gs, gs) * modulate_color;
+								}
+
+							} else {
+								for (unsigned int i = 0; i < vertex_count; ++i) {
+									w[i] = modulate_color;
+								}
+							}
+						}
+
+						const std::vector<int> &side_indices = voxel.get_model_side_indices(side);
+						const unsigned int index_count = side_indices.size();
+
+						{
+							int i = arrays.indices.size();
+							arrays.indices.resize(arrays.indices.size() + index_count);
+							int *w = arrays.indices.data();
+							for (unsigned int j = 0; j < index_count; ++j) {
+								w[i++] = index_offset + side_indices[j];
+							}
+						}
+
+						index_offset += vertex_count;
+					}
+
+					// Inside
+					if (voxel.get_model_positions().size() != 0) {
+						// TODO Get rid of push_backs
+
+						const std::vector<Vector3> &positions = voxel.get_model_positions();
+						unsigned int vertex_count = positions.size();
+						const Color modulate_color = voxel.get_color();
+
+						const std::vector<Vector3> &normals = voxel.get_model_normals();
+						const std::vector<Vector2> &uvs = voxel.get_model_uv();
+
+						Vector3 pos(x - 1, y - 1, z - 1);
+
+						for (unsigned int i = 0; i < vertex_count; ++i) {
+							arrays.normals.push_back(normals[i]);
+							arrays.uvs.push_back(uvs[i]);
+							arrays.positions.push_back(positions[i] + pos);
+							// TODO handle ambient occlusion on inner parts
+							arrays.colors.push_back(modulate_color);
+						}
+
+						const std::vector<int> &indices = voxel.get_model_indices();
+						unsigned int index_count = indices.size();
+
+						for (unsigned int i = 0; i < index_count; ++i) {
+							arrays.indices.push_back(index_offset + indices[i]);
+						}
+
+						index_offset += vertex_count;
+					}
+				}
+			}
+		}
+	}
+}
+
```

Comment:
Please, don't do this. Don't copy/paste this entire function. Besides, you are forcing blocky mesher for every kind of terrain. I wrote a comment earlier explaining how it should be done. The meshes that were generated *already* need to be used and batched, not recomputed.